### PR TITLE
Add autocomplete functional on vacation request page

### PR DESCRIPTION
**Pull request by daisy23333**
Changes

- Add autocomplete functional on vacation request page.
- Fix the vocation period on vacation-edit page.

### CODE REVIEW

File: sites/all/themes/vacation/scripts/carbon_copy.js [hunk 1]
```diff
@@ -67,5 +67,57 @@
...
+            icon: icon
+          });
+        });
+
+        $ele2.autocomplete({
+          minLength: 0,
+          source: data_list,
+          appendTo: ele3_string
+        })
+        .data( "ui-autocomplete" )._renderItem = function( ul, item ) {
```
Comment (jieyyal) [inline sites/all/themes/vacation/scripts/carbon_copy.js]: @daisy23333 Could we remove the extra white space in this function?

File: sites/all/themes/vacation/scripts/carbon_copy.js [hunk 2]
```diff
@@ -67,5 +67,57 @@
       removeCCUid($userItem.data('uid'));
       $userItem.remove();
     });
+
+    if ($('.page-vacation-request').length) {
```
Comment (jieyyal) [inline sites/all/themes/vacation/scripts/carbon_copy.js]: @daisy23333 Better to use another selector here, this is URL alias related. Can we use the form related selector?

File: sites/all/modules/custom/ciandt_wechat_vacation/ciandt_wechat_vacation.module [hunk 3]
```diff
@@ -442,7 +442,7 @@ function ciandt_wechat_vacation_form_vocation_node_form_alter(&$form, &$form_sta
   if (isset($user->name) && isset($options[$user->name])) {
     unset($options[$user->name]);
   }
-  $output = '<div class="popup"><div class="popup-top"><span class="close-btn">X</span></div><ul>';
+  $output = '<div class="popup"><div class="popup-top"><span class="close-btn">X</span></div><div class="approver-autocomplete"><input id="approver-autocomplete-search" class="form-text"/></div><ul class="initial-approver-list">';
```
Comment (jieyyal) [inline sites/all/modules/custom/ciandt_wechat_vacation/ciandt_wechat_vacation.module]: @daisy23333 Let's break the long code string to new line, Ex. `$output .= '<div class="approver-autocomplete">....';`

Comment (daisy23333) [reply sites/all/modules/custom/ciandt_wechat_vacation/ciandt_wechat_vacation.module]: @jieyyal  Fixed, please check.

File: sites/all/themes/vacation/scripts/carbon_copy.js (hunk 2 above)
Comment (daisy23333) [reply sites/all/themes/vacation/scripts/carbon_copy.js]: @jieyyal  Changed to form class "node-vocation-form" and move the function to a new js "vacation_autocomplete.js", please check.

File: sites/all/themes/vacation/scripts/carbon_copy.js (hunk 1 above)
Comment (daisy23333) [reply sites/all/themes/vacation/scripts/carbon_copy.js]: @jieyyal  Fixed, please check.

**Pull request merged**